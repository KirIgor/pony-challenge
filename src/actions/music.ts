import { ActionCreator } from 'redux';

// action types
export const TOGGLE_MUSIC = 'TOGGLE_MUSIC';
export type TOGGLE_MUSIC = typeof TOGGLE_MUSIC;

// actions

export interface ToggleMusic {
	type: TOGGLE_MUSIC;
}

export type MusicAction = ToggleMusic;

// action creators

export const toggleMusic: ActionCreator<ToggleMusic> = () => ({ type: TOGGLE_MUSIC });
