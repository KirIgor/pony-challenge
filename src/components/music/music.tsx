import * as React from 'react';

import './music.css';

interface Props {}

const Music = React.memo(({  }: Props) => {
	return (
		<div>
			<iframe src="music/silence.mp3" allow="autoplay" id="audio" style={{ display: 'none' }} />
			<audio id="music" autoPlay src="music/bundle.mp3" />
		</div>
	);
});

export default Music;
