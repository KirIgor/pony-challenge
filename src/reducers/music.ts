import { MusicState } from '../types/index';
import { MusicAction, TOGGLE_MUSIC } from '../actions/music';

export const defaultMusicState = new MusicState({
	playing: true
});

const musicReducer = (state: MusicState = defaultMusicState, action: MusicAction): MusicState => {
	switch (action.type) {
		case TOGGLE_MUSIC: {
			return state.update('playing', playing => {
				const res = !playing;

				if (res) {
					(document.getElementById('music') as HTMLAudioElement).play();
				} else {
					(document.getElementById('music') as HTMLAudioElement).pause();
				}

				return res;
			}) as MusicState;
		}
		default:
			return state;
	}
};

export default musicReducer;
