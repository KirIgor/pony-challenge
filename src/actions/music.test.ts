import { toggleMusic } from './music';

describe('music action creators', () => {
	describe('toggle action creator', () => {
		it('should dispatch correct action', () => {
			expect(toggleMusic()).toMatchSnapshot();
		});
	});
});
