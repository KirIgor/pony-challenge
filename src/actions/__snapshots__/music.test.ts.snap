// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`music action creators toggle action creator should dispatch correct action 1`] = `
Object {
  "type": "TOGGLE_MUSIC",
}
`;
